test/parallel/test-fs-read-stream-throw-type-error.js

```diff
@@ -18,10 +22,16 @@ const createReadStreamErr = (path, opt) => {
     {
       code: 'ERR_INVALID_ARG_TYPE',
       type: TypeError
-    });
+    }
+  );
 };
 
 createReadStreamErr(example, 123);
 createReadStreamErr(example, 0);
 createReadStreamErr(example, true);
 createReadStreamErr(example, false);
+
+// Should also throw on NaN (for https://github.com/nodejs/node/pull/19732)
+createReadStreamErr(example, { start: NaN });
```

Comment:
I think we can leave only `createReadStreamErr(example, { start: NaN })`, because if `start` is not passed, `end` is not defined (https://github.com/ryzokuken/node/blob/473dd3d03fe4d08445518f9ab8e2b78892c22e27/lib/fs.js#L2010).

And at least we need to delete `createReadStreamErr(example, { end: NaN })` to pass `node-test-commit-linux`.

Reply:
@anliting If `start` is not passed but `end` is, then the value is still set and type-checked: https://github.com/ryzokuken/node/blob/473dd3d03fe4d08445518f9ab8e2b78892c22e27/lib/fs.js#L2031-L2032

Reply:
If we are going to test the `end` part, we need have `start` to pass the range test first: `createReadStreamErr(example, { start: 0, end: NaN })`

Reply:
@anliting That would take a different branch, though?

Reply:
@addaleax 

Oh, I see that `end` is still set and type-checked now, my fault.

"If we are going to test the end part ..." means if we are going to test if this line work:
https://github.com/ryzokuken/node/blob/473dd3d03fe4d08445518f9ab8e2b78892c22e27/lib/fs.js#L2016

Reply:
We might need to change
```
  if (typeof this.end !== 'number' || Number.isNaN(this.end))
    this.end = Infinity;
```
to
```
  if (typeof this.end !== 'number')
    this.end = Infinity;
  else if (Number.isNaN(this.end))
    throw new ERR_INVALID_ARG_TYPE('end', 'integer', this.end);
```
to pass the `node-test-commit-linux`.

Reply:
@anliting Ah, right! Yes, we should probably do that.